qa/tasks/cephfs/test_admin.py

```diff
@@ -2047,6 +2046,79 @@ def test_fs_authorize(self):
                 errmsgs=self.EXPECTED_ERRMSG)
 
 
+class TestFSFail(TestAdminCommands):
+
+    MDSS_REQUIRED = 2
+    CLIENTS_REQUIRED = 1
+
+    def test_with_health_warn_oversize_cache(self):
+        health_warn = 'MDS_CACHE_OVERSIZED'
+        self.run_ceph_cmd('config set mds mds_cache_memory_limit 1K')
+        self.run_ceph_cmd('config set mds mds_health_cache_threshold 1.00000')
+
+        active_mds_id = self.fs.get_active_names()[0]
+        open_proc = self.mount_a.open_n_background(".", 4000)
+        self.wait_till_health_warn(health_warn, active_mds_id)
+
+        # actual testing begins now.
+        # test that "fs fail" fails without confirmation flag.
+        errmsg = 'mds_cache_oversized'
+        self.negtest_ceph_cmd(args=f'fs fail {self.fs.name}',
+                              retval=1, errmsgs=errmsg)
+        # test that "fs fail" passes with confirmation flag.
+        self.run_ceph_cmd(f'fs fail {self.fs.name} --yes-i-really-mean-it')
+
+    def test_with_health_warn_trim(self):
+        health_warn = 'MDS_TRIM'
+        active_mds_id = self.fs.get_active_names()[0]
+        open_proc = self.mount_a.open_n_background(".", 4000)
+        self.wait_till_health_warn(health_warn, active_mds_id)
+
+        # actual testing begins now.
+        # test that "fs fail" fails without confirmation flag.
+        errmsg = 'mds_trim'
+        self.negtest_ceph_cmd(args=f'fs fail {self.fs.name}',
+                              retval=1, errmsgs=errmsg)
+        # test that "fs fail" passes with confirmation flag.
+        self.run_ceph_cmd(f'fs fail {self.fs.name} --yes-i-really-mean-it')
+
+
+    def test_with_health_warn_trim(self):
+        health_warn = 'MDS_TRIM'
+        active_mds_id = self.fs.get_active_names()[0]
+        open_proc = self.mount_a.open_n_background(".", 4000)
+        self.wait_till_health_warn(health_warn, active_mds_id)
+
+        # actual testing begins now.
+        # test that "fs fail" fails without confirmation flag.
+        errmsg = 'mds_trim'
+        self.negtest_ceph_cmd(args=f'fs fail {self.fs.name}',
+                              retval=1, errmsgs=errmsg)
+        # test that "fs fail" passes with confirmation flag.
+        self.run_ceph_cmd(f'fs fail {self.fs.name} --yes-i-really-mean-it')
+
+
+    def test_with_health_warn_with_2_active_MDSs(self):
+        health_warn = 'MDS_TRIM'
+        mds1_id, mds2_id = self.fs.mds_ids
+        self.run_ceph_cmd(f'fs set {self.fs.name} max_mds 2')
```

Comment:
Tests in fa6a2a8f270bef22c903d18d1c9d56ae599f2033 use a mix of set_max_mds and `fs set <> max_mds`. Its confusing and the same is done here.

Reply:
I too find it confusing. Those helpers are really unnecessary. Switched to `fs set <> max_mds`.

Reply:
Better would be ti use set_max_mds(...)